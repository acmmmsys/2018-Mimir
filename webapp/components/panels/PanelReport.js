import React from 'react';
import { connect } from 'react-redux';
import { compose, nest } from 'recompose';
import { setTextField } from 'actions';

import Panel from 'layout/Panel'
import ColonoscopyReport from 'components/ui/ColonoscopyReport';

const enhance = compose(
    connect(
        state => ({
            attachedImages: state.report.attachedImages.map(imageId => state.app.images[imageId].source),
            patientName: state.report.patientName,
            dateOfBirth: state.report.dateOfBirth,
            generalPracticioner: state.report.generalPracticioner,
            hospitalNumber: state.report.hospitalNumber,
            dateOfProcedure: state.report.dateOfProcedure,
            endoscopist: state.report.endoscopist,
            nurses: state.report.nurses,
            medications: state.report.medications,
            instrument: state.report.instrument,
            extentOfExam: state.report.extentOfExam,
            visualization: state.report.visualization,
            coMorbidity: state.report.coMorbidity,
            indicationsForExamination: state.report.indicationsForExamination,
            procedurePerformed: state.report.procedurePerformed,
            findings: state.report.findings,
            endoscopicDiagnosis: state.report.endoscopicDiagnosis,
            recommendations: state.report.recommendations,
            followUp: state.report.followUp,
            opsc4Code: state.report.opsc4Code,
        }),
        dispatch => ({
            updateValue(field, value) {
                dispatch(setTextField(field, value));
            }
        })
    )
);

export default enhance(nest(Panel, ColonoscopyReport));