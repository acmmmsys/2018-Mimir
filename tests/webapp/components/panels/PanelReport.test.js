import React from 'react';
import PanelReport from '../../../../webapp/components/panels/PanelReport'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('PanelReport does render', () => {
  const wrapper = shallow(
      <PanelReport store={mockStore(initialState)}>This is a test!</PanelReport>
  );
  expect(wrapper).toMatchSnapshot();
});
