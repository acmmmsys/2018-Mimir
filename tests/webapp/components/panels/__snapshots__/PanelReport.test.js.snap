// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelReport does render 1`] = `
<nest(defaultProps(branch(branch(Panel))), ColonoscopyReport)
  attachedImages={Array []}
  coMorbidity="None"
  dateOfBirth="07.12.1993"
  dateOfProcedure="27.09.2012"
  endoscopicDiagnosis="Varices. Esophageal. Furthger 4 bands applied"
  endoscopist="Dan Brownman"
  extentOfExam="Second part of duodenum"
  findings="Polyps"
  followUp="High fiber diet"
  generalPracticioner="Billy Gregg"
  hospitalNumber="562346"
  indicationsForExamination="Survailence-Varices"
  instrument="G15 - H260"
  medications="Xylocaine Sprey"
  nurses="Jane Doe, Carol Smith"
  opsc4Code="G45 Gastroscopy (OGD)"
  patientName="Windom Earle"
  procedurePerformed="Gastoscopy (OGD)"
  recommendations="High fiber diet."
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  updateValue={[Function]}
  visualization="Good"
>
  This is a test!
</nest(defaultProps(branch(branch(Panel))), ColonoscopyReport)>
`;
